pkg/autodiscovery/autoconfig.go

```diff
@@ -495,6 +454,55 @@ func (ac *AutoConfig) pollConfigs() {
 	}()
 }
 
+// TODO: move to check scheduler
+func (ac *AutoConfig) processRemovedConfigs(removedConfigs []integration.Config) {
+	// Process removed configs first to handle the case where a
+	// container churn would result in the same configuration hash.
+	for _, config := range removedConfigs {
+		if !isCheckConfig(config) {
+			// skip non check configs.
+			continue
+		}
+		// unschedule all the checks corresponding to this config
```

Comment:
if I understand the code correctly, at this point, we're not sure that the config actually still has corresponding checks (thinking mostly of the case where this function is called from `processDelService`). Shouldn't cause any issue but it's worth pointing out in the comment I think.